dll/win32/shell32/folders/CRecycleBin.cpp

```diff
@@ -58,18 +58,57 @@ static const columninfo RecycleBinColumns[] =
 
 #define COLUMNS_COUNT  6
 
+BOOL WINAPI CBSearchRecycleBin(IN PVOID Context, IN HANDLE hDeletedFile);
+
+static PIDLRecycleStruct * _ILGetRecycleStruct(LPCITEMIDLIST pidl);
+
+typedef struct
+{
+    PIDLRecycleStruct *pFileDetails;
+    HANDLE hDeletedFile;
+    BOOL bFound;
+} SEARCH_CONTEXT, *PSEARCH_CONTEXT;
+
 /*
  * Recycle Bin folder
  */
 
-HRESULT CRecyclerExtractIcon_CreateInstance(LPCITEMIDLIST pidl, REFIID riid, LPVOID * ppvOut)
+HRESULT CRecyclerExtractIcon_CreateInstance(
+    LPCITEMIDLIST pidl, REFIID riid, LPVOID * ppvOut)
 {
+    PIDLRecycleStruct *pFileDetails = _ILGetRecycleStruct(pidl);
+
+    // Try to obtain the file
+    SEARCH_CONTEXT Context;
+    Context.pFileDetails = pFileDetails;
+    Context.bFound = FALSE;
+
+    if (pFileDetails == NULL) goto error;
+
+    EnumerateRecycleBinW(NULL, CBSearchRecycleBin, (PVOID)&Context);
+    if (Context.bFound)
+    {
+        // This should happen any time
```

Comment:
What should happen?

Reply:
The execution of that block, not finfing the file spots a bug on recycle bin implementation

Reply:
So done